code/modules/antagonists/eldritch_cult/eldritch_magic.dm

```diff
@@ -0,0 +1,469 @@
+	if(!.)
+		return FALSE
+	if(!istype(target,/mob/living/carbon/human))
+		if(!silent)
+			to_chat(user, "<span class='warning'>You are unable to touch [target]!</span>")
+		return FALSE
+	return TRUE
+
+/obj/effect/proc_holder/spell/pointed/touch/mad_touch/cast(list/targets, mob/user)
+	. = ..()
+	for(var/mob/living/carbon/target in targets)
+		if(ishuman(targets))
+			var/mob/living/carbon/human/tar = target
+			if(tar.anti_magic_check())
+				tar.visible_message("<span class='danger'>Spell bounces off of [target]!</span>","<span class='danger'>The spell bounces off of you!</span>")
+				return
+		if(target.mind && !target.mind.has_antag_datum(/datum/antagonist/e_cult))
+			SEND_SIGNAL(target, COMSIG_ADD_MOOD_EVENT, "gates_of_mansus", /datum/mood_event/gates_of_mansus)
+
+/obj/effect/proc_holder/spell/pointed/ash_final
+	name = "Nightwatcher's Rite"
+	desc = "Powerful spell that releases 5 streams of fire away from you."
+	school = "transmutation"
+	invocation = "F'RE"
+	invocation_type = "whisper"
+	charge_max = 300
+	range = 15
+	clothes_req = FALSE
+	action_icon = 'icons/mob/actions/actions_ecult.dmi'
+	action_icon_state = "flames"
+	action_background_icon_state = "bg_ecult"
+
+/obj/effect/proc_holder/spell/pointed/ash_final/cast(list/targets, mob/user)
+	for(var/X in targets)
+		var/T
+		T = line_target(-25, range, X, user)
+		INVOKE_ASYNC(src, .proc/fire_line, user,T)
+		T = line_target(10, range, X, user)
+		INVOKE_ASYNC(src, .proc/fire_line, user,T)
+		T = line_target(0, range, X, user)
+		INVOKE_ASYNC(src, .proc/fire_line, user,T)
+		T = line_target(-10, range, X, user)
+		INVOKE_ASYNC(src, .proc/fire_line, user,T)
+		T = line_target(25, range, X, user)
+		INVOKE_ASYNC(src, .proc/fire_line, user,T)
+	. = ..()
+
+/obj/effect/proc_holder/spell/pointed/ash_final/proc/line_target(offset, range, atom/at , atom/user)
+	if(!at)
+		return
+	var/angle = ATAN2(at.x - user.x, at.y - user.y) + offset
+	var/turf/T = get_turf(user)
+	for(var/i in 1 to range)
+		var/turf/check = locate(user.x + cos(angle) * i, user.y + sin(angle) * i, user.z)
+		if(!check)
+			break
+		T = check
+	return (getline(user, T) - get_turf(user))
+
+/obj/effect/proc_holder/spell/pointed/ash_final/proc/fire_line(atom/source, list/turfs)
+	var/list/hit_list = list()
+	for(var/turf/T in turfs)
+		if(istype(T, /turf/closed))
+			break
+
+		for(var/mob/living/L in T.contents)
+			if(L.anti_magic_check())
+				L.visible_message("<span class='danger'>Spell bounces off of [L.real_name]!</span>","<span class='danger'>The spell bounces off of you!</span>")
+				continue
+			if(L in hit_list || L == source)
+				continue
+			hit_list += L
+			L.adjustFireLoss(20)
+			to_chat(L, "<span class='userdanger'>You're hit by [source]'s fire breath!</span>")
+
+		new /obj/effect/hotspot(T)
+		T.hotspot_expose(700,50,1)
+		// deals damage to mechs
+		for(var/obj/mecha/M in T.contents)
+			if(M in hit_list)
+				continue
+			hit_list += M
+			M.take_damage(45, BURN, "melee", 1)
+		sleep(1.5)
```

Comment:
Does this work?

Reply:
yes, this is basically how ashdrakes make their flames, a for loop with a sleep()